### PR TITLE
raise setuptools minimum version and remove cap

### PR DESCRIPTION
**Pull request by dhellmann**
The setuptools issue introduced in 72.0.0 has been fixed in later
releases, so move our minimum past the bad release series and remove the
cap to allow newer versions.

### CODE REVIEW

**Review by shubhbapna — CHANGES_REQUESTED** [review]
We should also remove the setuptools constraint from the e2e test https://github.com/python-wheel-build/fromager/blob/main/e2e/constraints.txt

**Comment by dhellmann** [comment]
> We should also remove the setuptools constraint from the e2e test https://github.com/python-wheel-build/fromager/blob/main/e2e/constraints.txt

Done

File: e2e/constraints.txt
[the remark this answers is not in the document]
Comment (dhellmann) [reply e2e/constraints.txt:1]: I thought about that and decided to leave it in place to make it easier to add a constraint later if we need one. I could add a comment to that effect? Unless you think it's better to just remove it entirely.

[the remark this answers is not in the document]
Comment (shubhbapna) [reply e2e/constraints.txt:1]: comment works as well

[the remark this answers is not in the document]
Comment (dhellmann) [reply e2e/constraints.txt:1]: Done

**Review by shubhbapna — APPROVED** [review]

**Pull request merged**